client/health/health.go

```diff
@@ -163,6 +171,8 @@ func (monitor *Monitor) doHTTPCheck(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+	req.Header = monitor.header
+	req.Host = monitor.host
```

Comment:
I have some misunderstandings about the configuration of the host in K8s. We should not add the host parameter here, but instead actively extract the host field from the header and assign it to req.Host.

```go
 req.Host = monitor.header.Get("Host")
```